src/api/layer.js

```diff
@@ -100,13 +130,24 @@ export default class Layer {
      * @instance
      * @api
      */
-    setSource(source) {
+    async setSource(source) {
         this._checkSource(source);
+        const style = this._style;
+        if (style) {
+            var metadata = await source.requestMetadata(style);
+        }
+        if (this._style !== style) {
+            throw new Error('A style change was made before the metadata was retrieved, therefore, metadata is stale and it cannot be longer consumed');
```

Comment:
This proves that this method is dangerous and we must want to just have one method for updating the `source` and the `style`.

After talking it offline with @davidmanzanares, the proposal here is to remove this method and have an `update(source, style)` one. That method will be the same as the `setAtomicSourceStyle` in here, but it will allow you to just pass the `source` and optionally the `style`, the current `style` will be re-used if not passed, but we will keep the operation atomic avoiding this problem.

@IagoLast and @Jesus89, your thoughts?

Reply:
- We can use this API but better with an object because this forces you to pass always the same source if you want to update only the style.
```js
layer.update({
  source: ...,
  style: ...
});
```
- What happens with setStyle and blendToStyle?

Reply:
In carto.js we got the same problem, a style update could be fired before a second one was processed. 

So we made [those methods](https://carto.com/documentation/cartojs/docs/#cartostylecartocsssetcontent) async making the user responsible for the concurrency, so if you want to perform several operations one after another you have to wait for a promise first.

```javascript
await layer.setStyle(style0);
await layer.setSource(source0);
await layer.setStyle(style2);
```

Reply:
I'm OK with Iago's proposal, but sometimes you cannot set the style and later set the source. Imagine that you go from a points source to a polygons one. So we still need a method to update both _at the same time_.

About
> ```js
> layer.update({
>  source: ...,
>  style: ...
>});
>```

The `setStyle` would be still present, so you would use `update` when you need to update both or just the source. I don't like to have `options` API in these cases.

So, I guess our best approach is to keep `setSource` and `setStyle`, and introduce that new `setAtomicSourceStyle`, but I propose to rename it to `update`.

Reply:
Renamed